### PR TITLE
EAHW 2428: Manual promote to prod pipeline

### PR DESCRIPTION
**Pull request by daleelR**
- This PR automates the test deployment to run straight after dev deploys succesfully
- Allows manually promoting to prod only for succesful builds
  - Ensures promotion is only done on main branch and for prod env
  - Uses drone cli to check the last succesful build for this branch is the one we're promoting (from the last 100 builds on the repo)

### CODE REVIEW

**Review by JWeston-HO — DISMISSED** [review]

File: helm/values/prod-values.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: callisto-timecard-restapi
+
+service:
+  name: callisto-timecard-restapi
+  port: 3000
+
+replicaCount: 1
```
Comment (taomoh) [inline helm/values/prod-values.yaml:7]: Not an immediate concern, but at some point we would need this number to increased to foster availability/scalability in production?

File: pipeline-scripts/check-build-promotion.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+LAST_BUILD=$(
+  drone build ls --format '{{.Number}}' --branch "$DRONE_BRANCH" --status success \
+  --limit 100 "$DRONE_REPO" | head -n 1
+)
+if [ $LAST_BUILD != $DRONE_BUILD_PARENT ]; then
+  echo 'Could not promote as last build: ' $LAST_BUILD ' was not successful';
```
Comment (joannafawl) [inline pipeline-scripts/check-build-promotion.sh:6]: Should we reword this? $LAST_BUILD will return the last successful build, so the message is misleading

**Review by claire-guest — APPROVED** [review]

File: helm/values/prod-values.yaml (hunk 1 above)
Comment (daleelR) [reply helm/values/prod-values.yaml:7]: Do you think we should have a ticket created or something to bear in mind?

Comment (taomoh) [reply helm/values/prod-values.yaml:7]: I think that would be a good idea. We can easily forget about it. The architecture team could then feed into that decision as well. Let me know once the ticket is created and I will solve this comment

**Pull request merged**